vertical-pod-autoscaler/recommender/signals/prometheus_util.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+package signals
+
+import (
+	"fmt"
+	"io"
+	"encoding/json"
+	"time"
+)
+
+// Helper types used for parsing json returned by Prometheus.
+// This is the top-level structure of the response.
+type responseType struct {
```

Comment:
Add an information that prometheus didn't make this api available in form of any go library. 
https://github.com/prometheus/prometheus/blob/2d73d2b892853e95dbf157561e9df56ac220875e/web/api/v1/api.go#L92